File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):   
        self.rows = []
        self.columns = []

        # Obtengo las rows
        rows_str = matrix_string.split('\n')
        for row_str in rows_str:
            self.rows.append(list(map(int, row_str.split())))

        # Obtengo las columns
        for i in range(len(self.rows[0])):
            col = []
            for row in self.rows:
                col.append(row[i])
            self.columns.append(col)
            

    def row(self, index):
        return self.rows[index-1]

    def column(self, index):
        return self.columns[index-1]


#string = "1 2\n3 4\n5 6"
#print(string)
#print()
#m = Matrix(string)
#print(m.row(1))
#print(m.row(2))
#print(m.row(3))
#print(m.column(1))
#print(m.column(2))

File: javascript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(str) {
    this._rows = [];
    this._columns = [];
    
    let list_rows = str.split('\n');
    list_rows.forEach(element => {
      this._rows.push(element.split(' ').map(Number));
    });

    for(let i=0; i<this._rows[0].length; i++){
      let col = [];
      this._rows.forEach(element => {
        col.push(element[i]);
      });
      this._columns.push(col);
    }
  }

  get rows() {
    return this._rows; //.filter(value => !Number.isNaN(value));
  }

  get columns() {
    return this._columns;
  }
}
